polkadot/runtime/rococo/build.rs

```diff
@@ -14,12 +14,22 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
+use std::env;
 use substrate_wasm_builder::WasmBuilder;
 
+// note: needs to be synced with rococo-runtime-constants::time hard-coded string literal in
+// prod_or_fast macro.
+const ROCOCO_EPOCH_DURATION_ENV: &str = "ROCOCO_EPOCH_DURATION";
+
 fn main() {
-	WasmBuilder::new()
-		.with_current_project()
-		.import_memory()
-		.export_heap_base()
-		.build()
+	let mut builder = WasmBuilder::new().with_current_project().import_memory().export_heap_base();
+
+	if env::var(ROCOCO_EPOCH_DURATION_ENV).is_ok() && env::var("CARGO_FEATURE_FAST_RUNTIME").is_ok()
+	{
+		builder = builder.enable_feature("fast-runtime")
```

Comment:
The wasm builder automatically detects enabled features and passes them to the wasm build.